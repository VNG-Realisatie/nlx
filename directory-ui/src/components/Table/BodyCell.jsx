import { node, oneOf } from 'prop-types'
import styled from 'styled-components'

const BodyCell = styled.td`
    display: table-cell;
    padding: .6rem;

    font-size: 1rem;
    line-height: 1rem;
    font-weight: 400;
    
    text-align: ${p => p.align};
`

BodyCell.propTypes = {
  children: node,
  align: oneOf(['left', 'center', 'right'])
}

BodyCell.defaultProps = {
  align: 'left',
}

export default BodyCell
