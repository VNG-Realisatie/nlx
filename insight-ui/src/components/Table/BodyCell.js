// Copyright Â© VNG Realisatie 2020
// Licensed under the EUPL
//
import { node, oneOf } from 'prop-types'
import styled, { css } from 'styled-components'

const BodyCell = styled.td`
  background-color: #ffffff;
  display: table-cell;
  padding: ${(p) => (p.padding === 'none' ? 0 : '10px 16px 10px 16px')};

  font-size: 14px;
  line-height: 22px;
  font-weight: 400;

  text-align: ${(p) => p.align};

  ${(p) =>
    p.border === 'left' &&
    css`
      border-left: 1px solid #f0f2f7;
    `}

  ${(p) =>
    p.border === 'right' &&
    css`
      border-right: 1px solid #f0f2f7;
    `}
`

BodyCell.propTypes = {
  children: node,
  align: oneOf(['left', 'center', 'right']),
  border: oneOf(['left', 'right']),
  padding: oneOf(['none']),
}

BodyCell.defaultProps = {
  align: 'left',
}

export default BodyCell
