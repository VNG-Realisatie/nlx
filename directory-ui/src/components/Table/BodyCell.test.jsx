import React from 'react'
import { shallow } from 'enzyme'
import BodyCell from './BodyCell'

const addToTable = tableBodyCell =>
  <table>{ tableBodyCell }</table>

it('should match snapshot', () => {
  expect(shallow(<BodyCell/>)).toMatchSnapshot()
})

it('should render child elements', () => {
  expect(shallow(<BodyCell>
    <tr>
      <td>Table body</td>
    </tr>
  </BodyCell>).contains(<tr>
    <td>Table body</td>
  </tr>)).toEqual(true)
})

it('should support left alignment', () => {
  const wrapper = addToTable(<BodyCell align="left" />)
  expect(shallow(wrapper)).toMatchSnapshot()
})

it('should support right alignment', () => {
  const wrapper = addToTable(<BodyCell align="right" />)
  expect(shallow(wrapper)).toMatchSnapshot()
})

it('should support center alignment', () => {
  const wrapper = addToTable(<BodyCell align="center" />)
  expect(shallow(wrapper)).toMatchSnapshot()
})
