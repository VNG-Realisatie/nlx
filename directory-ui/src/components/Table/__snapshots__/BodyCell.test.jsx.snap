// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<StyledComponent
  align="left"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    display: table-cell;
    padding: .6rem;

    font-size: 1rem;
    line-height: 1rem;
    font-weight: 400;
    
    text-align: ",
          [Function],
          ";
",
        ],
      },
      "defaultProps": Object {
        "align": "left",
      },
      "displayName": "styled.td",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "align": [Function],
        "children": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "td",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`should support center alignment 1`] = `
<table>
  <styled.td
    align="center"
  />
</table>
`;

exports[`should support left alignment 1`] = `
<table>
  <styled.td
    align="left"
  />
</table>
`;

exports[`should support right alignment 1`] = `
<table>
  <styled.td
    align="right"
  />
</table>
`;
